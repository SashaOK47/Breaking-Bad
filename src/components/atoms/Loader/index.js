import PropTypes from "prop-types";
import style from "./styles.module.scss";

const Loader = ({ small }) => {
  let loader = style.loader;
  if (small) {
    loader += ` ${style.small}`;
  }
  return <div className={loader}></div>;
};
Loader.propTypes = {
  small: PropTypes.bool,
};
export default Loader;
