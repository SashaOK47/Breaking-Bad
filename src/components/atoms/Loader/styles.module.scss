@import "../../../assets/styles/main.scss";
.loader {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100px;
  height: 100px;
  margin-top: -50px;
  margin-left: -50px;
  border-radius: 50%;
  border-top: 3px solid $colorFontStatus;
  border-right: 3px solid transparent;
  animation: loader 0.7s linear infinite;
}
@keyframes loader {
  to {
    transform: rotate(360deg);
  }
}
