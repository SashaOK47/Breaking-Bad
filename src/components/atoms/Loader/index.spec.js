import React from "react";
import Loader from "./index";

const setUp = (props) => shallow(<Loader {...props} />);
const small = true;

describe("Error component", () => {
  it("should render Loader component", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });
  it("class is added small", () => {
    const component = setUp({ small });
    expect(component).toMatchSnapshot();
  });
});
