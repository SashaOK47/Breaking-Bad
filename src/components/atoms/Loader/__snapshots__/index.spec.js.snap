// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component class is added small 1`] = `
<div
  className="loader small"
/>
`;

exports[`Error component should render Loader component 1`] = `
<div
  className="loader"
/>
`;
